### PR TITLE
Bump up the version to v0.2.2

### LINKED ISSUE #1551: [v0.2.2] Release Tracker
**ETA**: ~~Nov 3rd (Fri) - Nov 6th (Mon).~~ Nov 17th (Fri) - 19th (Sun).

## Major changes

* Extensive refactoring for better tensor parallelism & quantization support
* Changes in scheduler: from 1D flattened input tensor to 2D tensor
* Bump up to PyTorch v2.1 + CUDA 12.1
* New models: Yi, ChatGLM, Phi
* Added LogitsProcessor API
* Preliminary support for SqueezeLLM

## PRs to be merged before the release

- [x] CUDA 12 #1527
- [x] Yarn #1264, #1161
- [x] Phi model #1664 
- ~~[ ] Support embedding inputs #1265~~

### PR DESCRIPTION
**Pull request by WoosukKwon**

### CODE REVIEW

**Review by zhuohan123 — APPROVED** [review]
LGTM!

**Review by simon-mo — APPROVED** [review]
